test/Microsoft.SqlTools.ServiceLayer.Test/QueryExecution/Common.cs

```diff
@@ -96,6 +97,11 @@ public static Query GetBasicExecutedQuery()
         public static Query GetBasicExecutedQuery(QueryExecutionSettings querySettings)
         {
             ConnectionInfo ci = CreateTestConnectionInfo(new[] {StandardTestData}, false);
+
+            // Query won't be able to request a new query DbConnection unless the ConnectionService has a 
+            // ConnectionInfo with the same URI as the query, so we will manually set it
+            ConnectionService.Instance.OwnerToConnectionMap[ci.OwnerUri] = ci;
```

Comment:
This is unfortunately true and may have been a bad decision on my part. The intent was to fail early when no ConnectionInfo is present, because ConnectionInfo contains the details needed to create a connection string in a real non-test scenario. This works for now, but it could be changed to prevent the need to do this in the future.